src/coloring.rs

```diff
@@ -104,10 +137,23 @@ pub fn graph_greedy_color(
 ///     edge_colors = rx.graph_greedy_edge_color(graph)
 ///     assert edge_colors == {0: 0, 1: 1, 2: 0, 3: 1, 4: 0, 5: 1, 6: 2}
 ///
+///
+/// .. [1] Adrian Kosowski, and Krzysztof Manuszewski, Classical Coloring of Graphs,
+///     Graph Colorings, 2-19, 2004. ISBN 0-8218-3458-4.
 #[pyfunction]
-#[pyo3(text_signature = "(graph, /)")]
-pub fn graph_greedy_edge_color(py: Python, graph: &graph::PyGraph) -> PyResult<PyObject> {
-    let colors = greedy_edge_color(&graph.graph);
+#[pyo3(text_signature = "(graph, /, greedy_strategy=GreedyStrategy::Degree)")]
+#[pyo3(signature=(graph, /, greedy_strategy=GreedyStrategy::Degree))]
+pub fn graph_greedy_edge_color(
+    py: Python,
+    graph: &graph::PyGraph,
+    greedy_strategy: GreedyStrategy,
```

Comment:
We could also extend the greedy edge coloring function with preset fn. What do you think?

Reply:
I think that's a good idea, just for consistency if nothing else.

Reply:
This is also supported: one can specify an optional callback to `graph_greedy_edge_color` that presets some of the edge colors. I have added tests and updated the release note.